cmd/main.go

```diff
@@ -233,11 +233,8 @@ func main() {
 	switch Options.DeployTarget {
 	case deployment_type_k8s:
 
-		objectHandler = s3wrapper.NewS3Client(&Options.S3Config, log, versionHandler, isoEditorFactory)
-		if objectHandler == nil {
-			log.Fatal("failed to create S3 client")
-		}
-		createS3Bucket(objectHandler)
+		objectHandler = createStorageClient(Options.DeployTarget, Options.Storage, &Options.S3Config, Options.JobConfig.WorkDir, log, versionHandler, isoEditorFactory)
```

Comment:
You are calling the same line 2 times, you can probably just move it outside of this switch statement.
After all the parameters are the same in both cases.